Add xml and xsd files

mpozdeev opened the pull request:
producrs.xml file contains test data for goods.
producrs.xsd this is the schema to the products.xml

mpozdeev committed:
Add xml and xsd files

producrs.xml file contains test data for goods.
producrs.xsd this is the schema to the products.xml

IliaIsakhin commented on src/main/resources/products.xml:
```diff
@@ -0,0 +1,141 @@
+    <category>
+        <categoryname>Audio equipment</categoryname>
+        <subcategory>
+            <namesubcategory>mp3 players</namesubcategory>
```
Можно сделать так `<subcategory name="mp3 players">`

IliaIsakhin commented on src/main/resources/products.xml:
```diff
@@ -0,0 +1,141 @@
+            <product>
+                <manufacturer>Ritmux</manufacturer>
+                <model>mp001</model>                    <!-- AA999(two letters and three numbers).-->
+                <DateOfManufacture>12-01-2018</DateOfManufacture> <!-- (DD-MM-YYYY).-->
```
Лучше писать с маленькой буквы

IliaIsakhin replied:
Тем более раз выше начал так делать

mpozdeev committed:
Add products_2.xsd, file xml modified

products_2.xsd:
added use of attributes;
element  "DateOfManufacture" renamed by "dateOfManufacture".

products.xml:
elements "categoryname" and "namesubcategory" are transferred into elements "category" and "subcategory" (respectively) as attributes

mpozdeev committed:
Changes made

Project structure changes made
Files  "class" added: Product, Subcategory, Category, Products. 
Annotations have been added to all JAXB classes.

IliaIsakhin commented on src/main/java/parser/JaxbParser.java:
```diff
@@ -0,0 +1,25 @@
+        try {
+            JAXBContext jaxbContext = JAXBContext.newInstance(Products.class);
+            Unmarshaller unmarshaller = jaxbContext.createUnmarshaller();
+            String xmlFilePath = "F:\\develop\\Lab3_xml_json\\resources\\products.xml";
```
Абсолютный путь лучше никогда не использовать

mpozdeev committed:
Added: files and changes

Added StaxJaxbParser file;
Changes: Absolute paths replaced by relative paths in the file JaxbParser

mpozdeev committed:
Added: files and changes

Added: XmlValidator and XmlValidator2 files;
Changes: Some elements xml and xsd have been renamed. Now the names of all elements begin with a small letter

mpozdeev committed:
Add file

Add ObjToJsonParser

mpozdeev committed:
A lot of changes

Changing the structure of the project
Files added

IliaIsakhin commented on src/main/java/parser/JsonToXmlParser.java:
```diff
@@ -0,0 +1,38 @@
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        ObjectMapper xmlMapper = new XmlMapper();
+        String jsonString = "{\"nameCategory\":\"Audio equipment\",\"subcategoryList\":[{\"nameSubcategory\":\"mp3 players\",\"productList\":[{\"manufacturer\":\"Ritmux\",\"model\":\"mp001\",\"dateOfManufacture\":\"12-01-2018\",\"color\":\"Red\",\"price\":19.99,\"amount\":7},{\"manufacturer\":\"SomY\",\"model\":\"mp002\",\"dateOfManufacture\":\"12-11-2018\",\"color\":\"Blue\",\"price\":29.99,\"am ...
```
JSON в отдельный файл

IliaIsakhin commented on src/main/java/parser/MainParser.java:
```diff
@@ -0,0 +1,82 @@
+        StringBuilder stringBuilder = new StringBuilder();
+
+        try {
+            reader = new BufferedReader(new FileReader(fileName));
```
try with resources

mpozdeev committed:
JSON output to file implemented in MainParser;
MainParser used: try with resources

mpozdeev committed:
Fixed bugs finded by FindBugs
